src/llmq/signing_shares.cpp

```diff
@@ -1481,36 +1492,76 @@ void CSigSharesManager::BanNode(NodeId nodeId, PeerManager& peerman)
     });
     nodeState.requestedSigShares.Clear();
     nodeState.banned = true;
+    // Banning affects request routing and can create immediate work
+    NotifyWorker();
 }
 
 void CSigSharesManager::WorkThreadMain(CConnman& connman, PeerManager& peerman)
 {
-    int64_t lastSendTime = 0;
-
     while (!workInterrupt) {
+        // capture epoch at loop start to detect intervening notifications
+        uint64_t startEpoch = workEpoch.load(std::memory_order_acquire);
         RemoveBannedNodeStates(peerman);
 
         bool fMoreWork = ProcessPendingSigShares(peerman, connman);
         SignPendingSigShares(connman, peerman);
 
-        if (TicksSinceEpoch<std::chrono::milliseconds>(SystemClock::now()) - lastSendTime > 100) {
-            SendMessages(connman);
-            lastSendTime = TicksSinceEpoch<std::chrono::milliseconds>(SystemClock::now());
-        }
+        bool didSend = SendMessages(connman);
```

Comment:
I think we had a cooldown for `SendMessages()` for a reason but I don't remember why exactly 🙈  All `qsig*` messages contain vectors, so maybe waiting a bit allowed us to utilise them better. Or maybe we simply did not want to push messages too often. Anyway, we need to confirm that there is no regressing in this part.